### ISSUE TITLE
Annotation processing behaves differently depending on presence of @Override

### ISSUE THREAD

**Issue by nhoughto** [micronaut-projects/micronaut-core#5154]
Thought it might be worth raising this just to check if it is expected behaviour or not, I just had a strange problem where a Micronaut HTTP controller method wasn't being recognised, was returning an error when all the other routes/methods worked perfectly. The root cause was that I had an interface with the `@Post` etc annotations in it, and then a class implementing the methods with the `@Singleton` annotation on it so that it gets picked up by Micronaut for injection etc. So annotations that are normally present together are split between interface and concrete class.

The problem was that for one of these methods, the one that wasn't be recognised, I had omitted the @Override annotation so Micronaut wasn't adding any of the interface method annotations into the method annotation metadata created by the annotation processor. This is a subtle failure that everything _looks_ right and compiles etc, but was strange behaviour at runtime when things just don't work, it was even stranger for me as this route was `@Secured` so it was actually failing in the SecurityFilter because the `routeMatch` was empty... not just an obvious '404 route doesn't exist'.

I imagine maybe you use `@Override` as a way of signalling when you do and do not want the annotation processor to follow the inheritance structure, feels like a weird way of doing it though.. and if you did want to do that opt-out would probably be a better default rather than opt-ing in via `@Override`. 

If this is not expected behaviour I can whip up a repro project. 👍

**Comment by jameskleeh** [comment]
Basically we assume that if a method is overriding another, the `@Override` annotation should be present. This is a standard convention that imo is reasonable to expect everyone to follow. We are using the presence of that annotation to determine whether or not the annotation metadata from parent methods should be searched. 

You could argue that we shouldn't do that and should instead just determine if the method is overriding any other method but that wouldn't be the simplest task and using the annotation is a simple way of achieving that.

In short yes this is the expected behavior
